### PR TITLE
#4: react example and GitHub pages actions

### LINKED ISSUE #4: Move replit React sandbox to a folder in the playground
**Context**
- We have a repl, but it would be nice to statically host so we can use as part of CI. (Currently we have to boot up the repl before we use it)
- https://replit.com/@PixieBrixTodd/ReactSandbox

### PR DESCRIPTION
**Pull request by twschiller**
Closes #4

### CODE REVIEW

File: .github/workflows/gh-pages.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        # Copy the directory without the directory name, see: https://unix.stackexchange.com/a/180987
+        run: |
+          mkdir public
+          mkdir public/example && cp --recursive example/. public/example
+          mkdir public/react-example && cp --recursive react-example/build/. public/react-example
+
+      - uses: actions/upload-artifact@v2
+        with:
+          name: playground-site
+          path: public
```
Comment (fregante) [inline .github/workflows/gh-pages.yaml:31]: What for?

Comment (twschiller) [reply .github/workflows/gh-pages.yaml:31]: Makes it easier to debug any path issues in projects. Any reason not to include it?

Comment (fregante) [reply .github/workflows/gh-pages.yaml:31]: > Makes it easier to debug any path issues in projects

The gh-pages deployment pushes to the repo itself:

https://github.com/pixiebrix/playground/tree/gh-pages/react-example

It also appears to preserve the history: https://github.com/pixiebrix/playground/commits/gh-pages

> Any reason not to include it

If the same data is available on the repo, this basically only slows down the deployment (more and more over time).

**Comment by twschiller** [comment]
> The gh-pages deployment pushes to the repo itself

The deploy action will only run on the `main` branch, though. Including the upload artifact in the workflow provides some reproducibility/debugging during PR development 

I'm OK removing it for now and can re-add if we hit problems on PR review

**Comment by fregante** [comment]
> Including the upload artifact in the workflow provides some reproducibility/debugging during PR development

Makes sense! We could also potentially set up Netlify to automatically deploy every PR. I know Vercel is not free for GitHub organizations, but I think Netlify is.

**Pull request merged**